### DISCUSSION TITLE
Running Or-Tools Java on Solaris Platform

### DISCUSSION THREAD

**Discussion by ahmtsn** [google/or-tools#3683] [Build and configuration issues] [answered]
Hello,

I would like to run my Google OR Tools java project on Solaris platform.

Currently, I develop my java project on a windows machine. In my java project; `ortools-java-*.jar` and `ortools-win32-x86-64-*.jar` libraries exist. Project runs normally on windows machine.

After building my project, I need to copy `dist` folder to Solaris platform (Solaris 11.3 SPARC). Without proper library I get the error with message `Resource ortools-sunos-sparcv9/ was not found in Classloader sun.misc.Launcher$AppClassLoader@...`.

How should I approach the situation?

Sincerely.

**Comment by lperron** [comment, marked as answer]
build from source, patch the maven config file to support solaris.

Good luck. you are on your own.

**Comment by Mizux** [comment]
Good news JNA seems to support/detect correctly your host `ortools-sunos-sparcv9/`,
Bad news we don't support `sunos` nor `sparcv9` BUT you can hack the cmake build system to make it working

Mostly here:
https://github.com/google/or-tools/blob/e106e236003d7db6f889bd64931c42d7f1cb6ee2/cmake/java.cmake#L59-L76

If I were you, I would try first to have a working example using https://github.com/mizux/java-native to have fast iteration
see: https://github.com/Mizux/java-native/blob/main/cmake/java.cmake
Then backport your changes to google/or-tools (both projects should use the same build system layout)
